src/lib.rs

```diff
@@ -246,12 +271,24 @@ mod tests {
                 e: &mut e,
                 f: &mut f,
                 g: &mut g,
-            }.with_common_anode();
+            }
+            .with_common_anode();
 
-            seven_segment.set(digit);
+            seven_segment.set(digit).expect("unable to set digit");
```

Comment:
The expect branch is unreachable. It can be explicitly marked by using  `.unwrap_or_else(|e| match e {})` instead of `.expect(...)`

[`unwrap_infallible`](https://docs.rs/unwrap-infallible/0.1.5/unwrap_infallible/) would be even more clear, if you care to add new dev-dependency.

Reply:
What is wrong here? Like, I know that the call will never `Err`, but how does it matter? I'm fine with adding both of your proposed solutions but I fail to see how the handling of an error which never occurs is relevant? Especially as `.expect()` is a rather well known expression, while I've never ever seen `.unwrap_or_else(|e| match e {})` before.

Reply:
It's just more clear in the code to a reader that it will never panic. Not wrong. `unwrap_infallible` is definitely more readable than match obviously. It's a shame core doesn't have it already.